src/vm/codeman.cpp

```diff
@@ -1302,121 +1303,128 @@ void EEJitManager::SetCpuInfo()
-        //    SSE4.2 - ECX bit 20  (buffer[10] & 0x10)
-        //    POPCNT - ECX bit 23  (buffer[10] & 0x80)
-        // CORJIT_FLAG_USE_AVX if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    OSXSAVE - ECX bit 27 (buffer[11] & 0x08)
-        //    AVX - ECX bit 28     (buffer[11] & 0x10)
-        // CORJIT_FLAG_USE_FMA if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    FMA - ECX bit 12     (buffer[9]  & 0x10)
-        // CORJIT_FLAG_USE_AVX2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    AVX2 - EBX bit 5     (buffer[4]  & 0x20)
-        // CORJIT_FLAG_USE_AVX_512 is not currently set, but defined so that it can be used in future without
-        // CORJIT_FLAG_USE_BMI1 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    BMI1 - EBX bit 3     (buffer[4]  & 0x08)
-        // CORJIT_FLAG_USE_BMI2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    BMI2 - EBX bit 8     (buffer[5]  & 0x01)
-        // CORJIT_FLAG_USE_LZCNT if the following feature bits are set (input EAX of 80000001H)
-        //    LZCNT - ECX bit 5    (buffer[8]  & 0x20)
-        // synchronously updating VM and JIT.
+
         (void) getcpuid(1, buffer);
-        // If SSE2 is not enabled, there is no point in checking the rest.
-        // SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
+
+        // If SSE/SSE2 is not enabled, there is no point in checking the rest.
+        //   SSE  is bit 25 of EDX   (buffer[15] & 0x02)
+        //   SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
         // TODO: Determine whether we should break out the various SSE options further.
-        if ((buffer[15] & 0x04) != 0)               // SSE2
+        if ((buffer[15] & 0x06) == 0x06)                                    // SSE & SSE2
         {
-            if ((buffer[8]  & 0x01) != 0)           // SSE3
-            {
-                CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE3);
-                if ((buffer[9]  & 0x02) != 0)           // SSSE3
-                {
-                    CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSSE3);
-                    if ((buffer[10] & 0x08) != 0)           // SSE4.1
-                    {
-                        CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE41);
-                    }
-                    if ((buffer[10] & 0x10) != 0)           // SSE4.2
-                    {
-                        CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_SSE42);
-                        if ((buffer[10] & 0x80) != 0)       // POPCNT
-                        {
-                            CPUCompileFlags.Set(CORJIT_FLAGS::CORJIT_FLAG_USE_POPCNT);
-                        }
-                    }
-                }
-            }
-            
-            if ((buffer[11] & 0x01) != 0)           // AESNI
```

Comment:
This check was incorrect. It should have been `if ((buffer[11] & 0x02) != 0)` to check bit 25 of ECX

Reply:
How was that not causing problems? Is it coincidentally the case that they are usually set together?

Reply:
I believe it wasn't causing problems because AES instructions are only used as part of HWIntrinsics, and that ISA hasn't been implemented yet.

Reply:
Ah, of course. Thanks.